python/tank/util/metrics.py

```diff
@@ -324,16 +327,95 @@ def _dispatch(self, metrics):
+    event name to the constructor:
+
+    Optionally, you can add your own specific metrics by using the
+    `properties` parameter. The latter simply takes a standard
+    dictionary.
+
+    The class also define numerous standard definition.
+    We highly recommand usage of them. Below is a complete typical usage:
+
+    ```
+    metric = EventMetric.log(EventMetric.GROUP_APP,
+        "User Logged In",
+        properties={
+            EventMetric.KEY_ENGINE: "tk-maya",
+            EventMetric.KEY_ENGINE_VERSION: "v0.2.2",
+            EventMetric.KEY_HOST_APP: "Maya",
+            EventMetric.KEY_HOST_APP_VERSION: "2017",
+            EventMetric.KEY_APP: "tk-multi-publish2",
+            EventMetric.KEY_APP_VERSION: "v0.2.3",
+            "CustomBoolMetric": True,
+            "RenderJobsSumitted": 173,
+        }
+    )
+    ```
+    """
 
-    def __init__(self, data):
-        """Initialize the object with a dictionary of metric data.
-        
-        :param dict data: A dictionary of metric data.
-        
+    # Event groups
+    GROUP_APP = "App"
+    GROUP_TASKS = "Tasks"
+    GROUP_MEDIA = "Media"
+    GROUP_TOOLKIT = "Toolkit"
+    GROUP_NAVIGATION = "Navigation"
+    GROUP_PROJECTS = "Projects"
+
+    # Event property keys
+    KEY_ACTION_TITLE = "Action Title"
+    KEY_APP = "App"
+    KEY_APP_VERSION = "App Version"
+    KEY_COMMAND = "Command"
+    KEY_ENGINE = "Engine"
+    KEY_ENGINE_VERSION = "Engine Version"
+    KEY_ENTITY_TYPE = "Entity Type"
+    KEY_HOST_APP = "Host App"
+    KEY_HOST_APP_VERSION = "Host App Version"
+    KEY_PUBLISH_TYPE = "Publish Type"
+
+    def __init__(self, group, name, properties=None):
+        """
+        Initialize a metric event using the specified parameters. Use this helper class
+        to create a suitable metric structure to be used with the
+
+        :param str group: A group or category this metric event falls into.
+        Although any values can be used, we encourage usage of the GROUP_*
+        definitions above.
+
+        :param: str name: A short descriptive event name or performed action.
+            The complete list can be found in the 'Shotgun Event Taxonomy' document.
+            Below are a few examples:
+                'Viewed Login Page'
+                'Logged In'
+                'Created Project'
+                'Toggled Project Favorite'
+                'Edited Task Status'
+                'Read Inbox Item'
+
+        :param: dict properties: An optional dictionary of extra properties to be attached to the metric event.
         """
-        self._data = data
+
+        """
+        We also add an empty event property dictionary that will be populated
+        with either or both specified properties or add system info properties.
+        """
+        self._data = {
+            "event_group": str(group),
+            "event_name": str(name),
+            "event_property": properties
+        }
+
+    def __repr__(self):
```

Comment:
FYI, there are some references still in the code (see line `84`) to the old metric objects. We should remove those. 

Those references are about the implementation of `__repr__` with respect to `log_once`. I'm trying to reason in my head if the event properties should be part of the "uniqueness" of a metric. If so, then we'll need to include them in the `__repr__` output in order to make the `log_once` function properly, or find some other way to check for uniqueness. Let me ponder it some more.

Reply:
Maybe I can add a timestamp class member that isn't part of actual payload or implement the hash() method taking properties in consideration.